source/github-helpers/get-pr-info.ts

```diff
@@ -1,24 +1,55 @@
+import * as pageDetect from 'github-url-detection';
+
 import * as api from './api';
 import {getConversationNumber} from '.';
 
 type PullRequestInfo = {
-	// TODO: Probably can be used for #3863 and #4679
+	// TODO: Use this for `restore-file` when GHE supports `compare`
 	baseRefOid: string;
-
 	// https://docs.github.com/en/graphql/reference/enums#mergeablestate
 	mergeable: 'CONFLICTING' | 'MERGEABLE' | 'UNKNOWN';
 	viewerCanEditFiles: boolean;
 };
 
-export default async function getPrInfo(number = getConversationNumber()!): Promise<PullRequestInfo> {
+export default async function getPrInfo(base: string, head: string, number = getConversationNumber()!): Promise<PullRequestInfo | undefined> {
+	if (pageDetect.isEnterprise()) {
+		const {repository} = await api.v4(`
+			repository() {
+				pullRequest(number: ${number}) {
+					mergeable
+					viewerCanEditFiles
+				}
+			}
+		`);
+
+		const compare = await api.v3(`compare/${base}...${head}?page=10000`); // `page=10000` avoids fetching any commit information, which is heavy
+		if (compare.status !== 'diverged') {
+			return;
+		}
+
+		return repository.pullRequest;
+	}
+
 	const {repository} = await api.v4(`
 		repository() {
 			pullRequest(number: ${number}) {
 				baseRefOid
 				mergeable
 				viewerCanEditFiles
+				headRef {
+					compare(headRef: "${base}") {
+						status
+						behindBy
+						aheadBy
+					}
+				}
 			}
 		}
 	`);
+
+	if (repository.pullRequest.headRef.compare.status !== 'DIVERGED') {
+		return;
+	}
+
 	return repository.pullRequest;
```

Comment:
This just mixes the logic of a specific feature with this generic `get pr info` function, which could return the PR info regardless of the need to update the PR.

I suggested returning `diverged: true` for both API calls, i.e. something like

```suggestion
	return {
		...repository.pullRequest,
		diverged: repository.pullRequest.headRef.compare.status === 'DIVERGED'
	};
```

And the logic just checks `pr.diverged`, or better yet `pr.needsUpdate` as a combination of `mergeable` and `diverged`


---

I suppose it's ok as is for now if it's tested, but it'll need to be changed eventually.  👍